request_a_govuk_domain/request/views.py

```diff
@@ -23,24 +23,52 @@
 
 class NameView(FormView):
     template_name = 'name.html'
-    form_class = NameForm
-    success_url = reverse_lazy('email')
 
-    def form_valid(self, form):
-        self.request.session['registration_data'] = {'registrant_full_name': form.cleaned_data['registrant_full_name']}
-        return super().form_valid(form)
+    def get(self, request):
+        params = {}
+        if 'change' in request.GET:
+            params['registrant_full_name'] = request.session['registration_data']['registrant_full_name']
+            form = NameForm(params)
+        else:
+            form = NameForm()
+        return render(request, self.template_name, {'form': form})
+
+    def post(self, request):
+        form = NameForm(request.POST)
+        if form.is_valid():
+            registration_data = request.session.get('registration_data', {})
+            registration_data['registrant_full_name'] = form.cleaned_data['registrant_full_name']
+            request.session['registration_data'] = registration_data
```

Comment:
Should we store the value if it is cancel?

Reply:
This way of doing confirmation pages is new (to me) and begs a few questions as to how the flow behaves and indeed how the data is modified as the user goes back and forth. I think we'll compile all our questions and put them to our designers.